src/perf/lib/PerfServer.cpp

```diff
@@ -71,6 +84,39 @@ PerfServer::Init(
         }
     }
 
+    if (TryGetVariableUnitValue(argc, argv, "delay", &DelayMicroseconds, nullptr)) {
+        const char* DelayTypeString = nullptr;
+        DelayType = SYNTHETIC_DELAY_FIXED;
+
+        if (TryGetValue(argc, argv, "delayType", &DelayTypeString)) {
+#ifndef _KERNEL_MODE
+            if (IsValue(DelayTypeString, "variable")) {
+                //
+                // Initialize variable delay parameters
+                //
+                DelayType = SYNTHETIC_DELAY_VARIABLE;
+                Lambda = ((double)1) / DelayMicroseconds;
+                MaxFixedDelayUs = static_cast<uint32_t>(4 * (uint64_t)DelayMicroseconds);
+                if (MaxFixedDelayUs < 1000) MaxFixedDelayUs = 1000;
+            } else if (!IsValue(DelayTypeString, "fixed")) {
+                WriteOutput("Failed to parse DelayType[%s] parameter. Using fixed DelayType.\n", DelayTypeString);
+            }
+#else
+            WriteOutput("Kernel mode supports fixed delay only\n");
+#endif // !_KERNEL_MODE
+        }
+
+        ProcCount = (uint16_t)CxPlatProcCount();
+        DelayWorkers = new (std::nothrow) DelayWorker[ProcCount];
+        for (uint16_t i = 0; i < ProcCount; ++i) {
+            if (!DelayWorkers[i].Initialize(this, i)) {
+                WriteOutput("Failed to init delay workers.\n");
+                return QUIC_STATUS_INTERNAL_ERROR;
+            }
+        }
+        DelayPoolUsed = TRUE;
```

Comment:
Is this variable strictly necessary? Why not just check if `DelayWorkers` is non-null?